src/golang.org/exercises/defang/defang.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+)
+
+func defangIPaddr(address string) string {
+	result := ""
+	for _, char := range address {
+		if string(char) == "." {
+			result += "[.]"
+		} else {
+			result += string(char)
```

Comment:
Using string concatenation makes this an o(n^2) algorithm (because each concatination requires copying the string to a new buffer, an amortized O(n) operation).  For n < 16 (all IPv4 addresses) this doesn't matter much.  For larger strings (or to preemptively quell comments like this) you can use techniques from:

https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go